lib/axon.ex

```diff
@@ -80,6 +80,50 @@ defmodule Axon do
   @doc false
   defstruct [:id, :name, :output_shape, :parent, :op, :params, :opts]
 
+  @doc """
+  A custom Axon layer.
+  """
+  def layer(parent, op, output_shape, parameters, name \\ nil, opts \\ [])
+      when is_atom(op) or is_function(op) do
+    op = if is_atom(op), do: op, else: :layer
```

Comment:
We need a bit more docs. :)

In any case, when is `op` an atom? Is that only for Axon own layers?

Reply:
Yes, but I think with this change I can greatly simplify the compiler which may mean that it is always a function or perhaps not, there are just a few things I need to think about first.

Reply:
Yeah, I think normalizing is good and then have the “op” as an option to optionally name it?